### PR TITLE
Add the ability to override config from REPLs

### PR DESCRIPTION
**Pull request by jeffhhk**

### CODE REVIEW

**Comment by gregr** [comment]
It looks like a bunch of other changes tagged along here.  Is there a way to reduce the scope of this PR?

**Comment by jeffhhk** [comment]
My bad.  Fixed now.

File: medikanren/configref.rkt [hunk 1]
```diff
@@ -14,30 +15,120 @@
...
+    (error "invalid configuration:" config))
+)
+;;; config-combine
+;; Default configs go last, and must contain a superset of all config keys.
+;; Could be faster, but intended to only be run once at startup.
+(define (config-combine . configs)
+  (define (find k configs)
+    (if (empty? configs)
+      (error "config defaults must contain a superset of all other config keys")
+      #f)
```
Comment (gregr) [inline medikanren/configref.rkt:50]:
```
(when (empty? configs)
  (error "config defaults must contain a superset of all other config keys"))
```

**Comment by gregr** [comment]
Looks fine to me.  Feel free to merge.

File: medikanren/configref.rkt (hunk 1 above)
Comment (jeffhhk) [reply medikanren/configref.rkt]: Now changed.

**Pull request merged**